jberet-se/src/main/java/org/jberet/se/BatchSEEnvironment.java

```diff
@@ -103,6 +126,35 @@ protected int getMaximumPoolSize() {
         this.jobXmlResolver = new ChainedJobXmlResolver(userJobXmlResolvers, DEFAULT_JOB_XML_RESOLVERS);
     }
 
+
+    /**
+     * This method will parse the value like: {@code ${DB_PASS:xyz}}. It will try to fetch the value of the
+     * environment variable {@code DB_PASS} firstly, and if it does not exist, the method will return the
+     * default value {@code xyz}.
+     *
+     * @param propVal The property value to parse.
+     * @return Get the parsed value of the {@code propVal}.
+     */
+    static String parseProp(String propVal) {
+        if (Pattern.matches(PROP_PATTERN_STR, propVal)) {
+            Matcher matcher = PROP_PATTERN.matcher(propVal);
```

Comment:
since there is already a compiled `Pattern`, we should use it instead of the literal regex. After the pattern compilation, the string value should not be used.  Maybe we can do something like this:

```
 Matcher matcher = PROP_PATTERN.matcher(propVal);
 if (matcher. matches()) {
   ...
  }
```